community/kernel/src/main/java/org/neo4j/kernel/impl/api/KernelTransactionImplementation.java

```diff
@@ -458,10 +460,20 @@ private void commit() throws TransactionFailureException
 
         try ( CommitEvent commitEvent = transactionEvent.beginCommitEvent() )
         {
-            // Trigger transaction "before" hooks
-            if ( (hooksState = hooks.beforeCommit( txState, this, storeLayer )) != null && hooksState.failed() )
+            // Trigger transaction "before" hooks.
+            if ( hasChanges() )
```

Comment:
I can imagine that it can be useful for monitoring purposes - when you track last commit/rollback transaction time or count number of successful or failed transactions